osdep/io.c

```diff
@@ -140,6 +140,22 @@ char *mp_to_utf8(void *talloc_ctx, const wchar_t *s)
     return ret;
 }
 
+char *format_windows_path(void *talloc_ctx, const char *path) {
+    char *filename_copy = talloc_strdup(talloc_ctx, path);
+    if (strlen(filename_copy) >= MAX_PATH && 
+        strncmp("\\\\?\\", filename_copy, 4) != 0) {
+        for (int i = 0; i < strlen(filename_copy); i++) {
```

Comment:
The loop condition `i < strlen(filename_copy)` does `strlen` on each iteration (each char of that string), which makes it O(N^2).  Big no.

Similar loop which is more correct would be (untested):
```c
char *s = filename_copy;
while ((s = strchr(s, '/'))
    *s++ = '\\';

```

There might also be some existing string replace function, but not sure about that.